### ISSUE TITLE
r10 branch make error

### ISSUE THREAD

**Issue by rekham1077** [jts/nanopolish#926]
Hello, 

I am trying to install the r10 branch and I get this error: 

make: *** [eigen/INSTALL] Error 8

I am trying to generate a new consensus for a canu assembly. I had tried the most recent version before the r10 branch, but the problem I encountered was that my vcf files were empty. It was suggested that I try the r10 branch because I used FLO-MIN111. 

I did not encounter installation issues with the main branch and I think I am taking the same steps as before. 

What should I try in order to fix this? Or is there another solution I could try?

**Comment by jts** [comment]
Hi,

The path to the EIGEN source code moved a while ago, and it seems this
branch doesn't have the fix.  I'll try to fix this tomorrow.

Jared

**Comment by rekham1077** [comment]
Okay thanks! I will test it.

**Comment by jts** [comment]
Did you forget the `--recursive` flag during `git clone`?

**Comment by rekham1077** [comment]
@jts Yes I think I did. But I just tried this: git clone --recursive --branch r10 https://github.com/jts/nanopolish.git

The installation works fine but then I get no variants in the vcf files again. This is what I tried:

python3 ~/nanopolish/scripts/nanopolish_makerange.py contigs_no_bubbles.fasta | parallel --results nanopolish.results -P 8 ~/nanopolish/nanopolish variants --consensus -o polished.{1}.vcf -w {1} -r YR015_all_reads.fastq -b reads_aln_canu_draft.bam -g contigs_no_bubbles.fasta -t 4 --min-candidate-frequency 0.1

**Comment by jts** [comment]
Did you get any errors or output printed to the terminal/log files?

**Comment by rekham1077** [comment]
@jts Yes, if I open the stderr files it says this : 

total reads: 897, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 897, bad fast5: 0

Its the same issue I had with the main branch. But, if I look in the bam file most of my reads are mapped with high enough quality (263283/341897 have mapq of at least 20).

**Comment by cstill8221992** [comment]
Hi @jts, I wanted to follow-up on this discussion as I'm having a similar high "no alignment" issue when trying to run call-methylation. I am within the r10 branch, and I'm using a FLO-MIN111 flowcell. An example of my code is below:

`nanopolish call-methylation -r fastq_files/basecalls.fastq -b winnowmap_derived/LRS_exp1_mapped.sorted.bam -g /home/ubuntu/genome_stuff/hg38_genome.fa -t 16 -q cpg --model=/home/ubuntu/nanopolish/etc/r10-models/r10_450bps.cpg.9mer.template.model --verbose > LRS_Exp1_meth_call.tsv`

The output has continuously been:
[post-run summary] total reads: 31854, unparseable: 0, qc fail: 0, could not calibrate: 114, no alignment: 31751, bad fast5: 0

A couple of notes:
- I have tried with and without the -q option
- Here I specify the model to use however got the same result when the model was not specified

Do you find anything inherently wrong with this command? In the issue #844 you mentioned about an issue with detection of the right model to use for the FLO-MIN111, however in later iterations of my command, I include the --model option in order to directly provide the model I want used.

Any help would be greatly appreciated.

Thanks,
Chris

**Comment by jts** [comment]
Hi,

The `--model` command tells nanopolish to load the given model, but it will still try to detect the pore type from the fast5 files. If it cannot detect the pore type, it will default to R9.4 and use a built-in model. This may be the problem. 

If you're comfortable changing the code and recompiling nanopolish you can try to hard-code the pore type to R10 here to see if it fixes the problem:

https://github.com/jts/nanopolish/blob/r10/src/nanopolish_squiggle_read.cpp#L104

Note that I haven't tested nanopolish on R10.4 yet.

Jared

**Comment by touala** [comment]
Hi Jared,

I'm also interested in processing R10.4 data. I tried messing around in your code but I don't think hard-coding the pore type to R10 is sufficient (even after avoiding the VBZ incompatibility and multi-threading issue, although the latest maybe only on my side). For `eventalign`, I see that the `qc_simple_event_alignment` failed for all reads with both `avg_log_emission` too low compared to `parameters.min_average_log_emission` and `spanned` being `false`. Maybe the model is not compatible anymore...

I'm mostly commenting to keep track of this issue but I hope you find the time to get around it. Thank you for all the work on nanopolish.

Alan

**Comment by hasindu2008** [comment]
@touala @cstill3928 @rekham1077 

if you are going to use nanoporelish for methcalling or eventalign of R10 data, the trained R10.4.1 models are available in [f5c](https://github.com/hasindu2008/f5c/). 

The training process and some benchmark results are [here](https://hasindu2008.github.io/f5c/docs/r10train) and [here](https://github.com/hasindu2008/f5c/releases/tag/v1.2.
